### DISCUSSION TITLE
Transparent with opacity

### DISCUSSION THREAD

**Discussion by romdim** [tailwindlabs/tailwindcss#8982] [Help] [answered]
Hi,

I cannot find a way to result in a color like `transparent 55%;`.
In particular my use-case is for creating a red strikethrough from left bottom corner to right top.
For that I use:
`bg-gradient-to-tl from-transparent/45 via-strikethrough-red to-transparent/55`
I've already set-up opacity for 45 & 55 and strikethrough-red in my config.
The from-transparent/45 for example result in `rgb(0 0 0 / 45%))` while I would expect it to be `transparent 45%`.
Is there something I am missing?

Thanks in advance!

**Comment by wongjn** [comment, marked as answer]
There is no way with default Tailwind classes to set color stops at specific points – the only ones are `0%` with `from-<color>`, `50%` with `via-<color>` and `100%` with `to-<color>` as per the [documentation](https://tailwindcss.com/docs/gradient-color-stops#middle-color). The class name you tried works as intended as the gradient stop classes are structured like `<from|via|to>-<color>` and thus `transparent/50%` is the `<color>` portion, using the [new color opacity syntax that was introduced in v3](https://tailwindcss.com/docs/upgrade-guide#new-opacity-modifier-syntax), see for example, [background color](https://tailwindcss.com/docs/background-color#changing-the-opacity).

You could write a custom class or plugin for your use case.

**Reply by romdim** [reply]
I got it that I cannot get other stops than 0, 50 & 100%. But my concern is about the `from-transparent/45` being translated to pretty much `from-black/45`, meaning `rgb(0 0 0 / 45%))` instead of `transparent 45%`.

**Reply by wongjn** [reply]
Ah, I think this is because the logic of Tailwind is that `transparent` is equivalent to `rgb(0 0 0 / 0)` but then if you modify it with an opacity to `45%`, it then transforms the alpha channel to `rgb(0 0 0 / 45%)`. This *is* an edge case but it shouldn't really matter I don't think, since `transparent` should be by definition have an opacity of `0%` so there is some misunderstanding of how colors work if you are trying to use `transparent` with an opacity. You can't define `transparent` with an alpha opacity value in CSS either – it's always only valid as a color value by itself as a keyword.

**Reply by romdim** [reply]
This is indeed where it goes wrong. The thing is that this is not an edge case, it's valid css where you can have transparent with opacity in order to form a specific background. In my case it is a red diagonal line striking through text. You can check my code below:

```
<div className="relative">
       <p>Text here</p>
        <div
            style={{background: "linear-gradient(to left top, transparent 45.5%, #d10813 48.5%, #d10813 51.5%, transparent 54.5%)"}}
            className="absolute top-0 w-full h-full"
            // bg-gradient-to-tl from-transparent/45 via-strikethrough-red to-transparent/55"
        ></div>
    </p>
```

I've used tailwindcss for the classnames and react. In the commented out line you can see the classes I tried to use from tailwind.
It's not possible though because of the limitation stated above.

**Reply by wongjn** [reply]
I do not see where you have used opacity on `transparent` in your example. I think you are getting confused with what `transparent 45.5%` or `transparent 54.5%` means; it means transparent at 45.5% and 54.5% gradient positions, not the opacity of `transparent`.
